src/components/forms/SearchBar/SearchBar.tsx

```diff
@@ -75,33 +76,35 @@ const SearchBar: React.FC<SearchBarProps> = ({
   };
 
   const searchInputRef = useRef<HTMLInputElement>(null);
+  const [query, setQuery] = useState(defaultValue);
   const [isSearching, setIsSearching] = useState(false);
   const [searchPerformed, setSearchPerformed] = useState(false);
   const isMounted = useMounted();
 
-  const performSearch = async (query) => {
+  const debouncedQuery = useDebounce(query, 500);
+
+  const maxLength = 100;
```

Comment:
I wouldn't introduce this validation in fix commit, also it's not a good idea to hardcode the maxLength.